src/test/AaveV2CollectorContractConsolidation.t.sol

```diff
@@ -0,0 +1,248 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.17;
+
+import "@forge-std/Test.sol";
+import "@forge-std/console.sol";
+import {AaveV2Ethereum} from "@aave-address-book/AaveV2Ethereum.sol";
+import {AaveV2CollectorContractConsolidation} from "../AaveV2CollectorContractConsolidation.sol";
+import {AggregatorV3Interface} from "../external/AggregatorV3Interface.sol";
+import {ERC20} from "@openzeppelin/token/ERC20/ERC20.sol";
+
+contract AaveV2CollectorContractConsolidationTest is Test {
+    event Swap(address indexed tokenIn, address indexed tokenOut, uint256 amountIn, uint256 amountOut);
+
+    address public constant USDC_WHALE = 0x55FE002aefF02F77364de339a1292923A15844B8;
+    address public constant ETH_WHALE = 0xF977814e90dA44bFA03b6295A0616a897441aceC;
+
+    ERC20 public constant USDC = ERC20(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+    address public constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+
+    AaveV2CollectorContractConsolidation public collectorContract;
+
+    address[18] public allTokens;
+
+    function setUp() public {
+        vm.createSelectFork(vm.rpcUrl("mainnet"), 16183428); // December 14, 2022
+
+        collectorContract = new AaveV2CollectorContractConsolidation();
+        allTokens[0] = collectorContract.ARAI();
+        allTokens[1] = collectorContract.AAMPL();
+        allTokens[2] = collectorContract.AFRAX();
+        allTokens[3] = collectorContract.FRAX();
+        allTokens[4] = collectorContract.AENS();
+        allTokens[5] = collectorContract.SUSD();
+        allTokens[6] = collectorContract.ASUSD();
+        allTokens[7] = collectorContract.TUSD();
+        allTokens[8] = collectorContract.ATUSD();
+        allTokens[9] = collectorContract.AMANA();
+        allTokens[10] = collectorContract.ADPI();
+        allTokens[11] = collectorContract.ABUSD();
+        allTokens[12] = collectorContract.BUSD();
+        allTokens[13] = collectorContract.ZRX();
+        allTokens[14] = collectorContract.AZRX();
+        allTokens[15] = collectorContract.MANA(); // Up to here tests
+        allTokens[16] = collectorContract.AUST();
+        allTokens[17] = collectorContract.ARENFIL();
+    }
+
+    function testSwapZeroAmount() public {
+        address frax = collectorContract.FRAX();
+        vm.expectRevert(AaveV2CollectorContractConsolidation.OnlyNonZeroAmount.selector);
+        collectorContract.swap(frax, 0);
+    }
+
+    function testSwapAllAAmpl() public {
```

Comment:
You should be able to collapse the test cases for each of these individual tokens into one test by looping through the finite list of assets within the test case. That probably makes more sense than to have 20 or so individual tests (i.e one for each asset)

Reply:
Yeah, I'm sanity checking everything and getting exact values and debugging each test individually before collapsing into one.